### ISSUE TITLE
Rust error

### ISSUE THREAD

**Issue by simonbcn** [bpetlert/pacman-mirrorup#2]
```
Arch Linux
pacman-mirrorup 0.4.0-1
```

```
$ env RUST_BACKTRACE=full /usr/bin/pacman-mirrorup -o kk -T 10
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', src/mirror.rs:146:64
stack backtrace:
0:     0x55f2531dbeb0 - <unknown>
1:     0x55f2531fd65c - <unknown>
2:     0x55f2531d63b5 - <unknown>
3:     0x55f2531ddf7b - <unknown>
4:     0x55f2531dda51 - <unknown>
5:     0x55f2531de654 - <unknown>
6:     0x55f2531de0f7 - <unknown>
7:     0x55f2531dc3ac - <unknown>
8:     0x55f2531de089 - <unknown>
9:     0x55f252e8dcf1 - <unknown>
10:     0x55f252e8dc3d - <unknown>
11:     0x55f252e9904d - <unknown>
12:     0x55f252eb4532 - <unknown>
13:     0x55f252ea11a3 - <unknown>
14:     0x55f252eb45cd - <unknown>
15:     0x55f252ea11a3 - <unknown>
16:     0x55f252eb45cd - <unknown>
17:     0x55f252ea11a3 - <unknown>
18:     0x55f252eb45cd - <unknown>
19:     0x55f252ea11a3 - <unknown>
20:     0x55f252eb45cd - <unknown>
21:     0x55f252eabe95 - <unknown>
22:     0x55f252e87171 - <unknown>
23:     0x55f253136ee6 - <unknown>
24:     0x55f2531392b5 - <unknown>
25:     0x55f25313a47b - <unknown>
26:     0x55f2531e1cb7 - <unknown>
27:     0x7f66b384d259 - start_thread
28:     0x7f66b362b5e3 - __GI___clone
29:                0x0 - <unknown>
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', src/mirror.rs:146:64
stack backtrace:
0:     0x55f2531dbeb0 - <unknown>
1:     0x55f2531fd65c - <unknown>
2:     0x55f2531d63b5 - <unknown>
3:     0x55f2531ddf7b - <unknown>
4:     0x55f2531dda51 - <unknown>
5:     0x55f2531de654 - <unknown>
6:     0x55f2531de0f7 - <unknown>
7:     0x55f2531dc3ac - <unknown>
8:     0x55f2531de089 - <unknown>
9:     0x55f252e8dcf1 - <unknown>
10:     0x55f252e8dc3d - <unknown>
11:     0x55f252e9904d - <unknown>
12:     0x55f252eb4532 - <unknown>
13:     0x55f252eabe95 - <unknown>
14:     0x55f252e87171 - <unknown>
15:     0x55f252ea1266 - <unknown>
16:     0x55f252eb45cd - <unknown>
17:     0x55f252ea11a3 - <unknown>
18:     0x55f252eb45cd - <unknown>
19:     0x55f252eabe95 - <unknown>
20:     0x55f252e87171 - <unknown>
21:     0x55f253136ee6 - <unknown>
22:     0x55f2531392b5 - <unknown>
23:     0x55f25313a47b - <unknown>
24:     0x55f2531e1cb7 - <unknown>
25:     0x7f66b384d259 - start_thread
26:     0x7f66b362b5e3 - __GI___clone
27:                0x0 - <unknown>
```

**Comment by msjr65** [comment]
confirming the same issue, I'm also on Arch Linux
pacman-mirrorup 0.4.0-1

**Comment by bpetlert** [comment]
This happen when program cannot fetch database file from a mirror. it should be fixed in commit ebd52f7.
Anyway I released new version 0.5.0. Please help test it.

**Comment by simonbcn** [comment]
It works now. Thanks.
